test/Microsoft.IdentityModel.Tokens.Tests/EncryptingCredentialsTests.cs

```diff
@@ -0,0 +1,198 @@
+using Microsoft.IdentityModel.Tests;
+using Xunit;
+
+#pragma warning disable CS3016 // Arrays as attribute arguments is not CLS-compliant
+
+namespace Microsoft.IdentityModel.Tokens.Tests
+{
+    public class EncryptingCredentialsTests
+    {
+        //public EncryptingCredentials(SecurityKey key, string alg, string enc)
+        [Theory, MemberData(nameof(ConstructorATheoryData))]
+        public void ConstructorA(EncryptingCredentialsTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.ConstructorA", theoryData);
+            var context = new CompareContext($"{this}.ConstructorA, {theoryData.TestId}");
+            try
+            {
+                var encryptingCredentials = new EncryptingCredentials(theoryData.Key, theoryData.Alg, theoryData.Enc);
+                theoryData.ExpectedException.ProcessNoException();
+            }
+            catch (Exception exception)
+            {
+                theoryData.ExpectedException.ProcessException(exception);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        //Used in scenarios when a key represents a 'shared' symmetric key
+        //public EncryptingCredentials(SecurityKey key, string enc)
+        [Theory, MemberData(nameof(ConstructorBTheoryData))]
+        public void ConstructorB(EncryptingCredentialsTheoryData theoryData)
+        {
+            TestUtilities.WriteHeader($"{this}.ConstructorB", theoryData);
+            var context = new CompareContext($"{this}.ConstructorB, {theoryData.TestId}");
+            try
+            {
+                var encryptingCredentials = new EncryptingCredentials(theoryData.Key, theoryData.Enc);
+                //Algorithm value should be 'None'
+                IdentityComparer.AreEqual(encryptingCredentials.Alg, SecurityAlgorithms.None, context);
+                theoryData.ExpectedException.ProcessNoException();
+            }
+            catch (Exception exception)
+            {
+                theoryData.ExpectedException.ProcessException(exception);
+            }
+
+            TestUtilities.AssertFailIfErrors(context);
+        }
+
+        public static TheoryData<EncryptingCredentialsTheoryData> ConstructorATheoryData()
+        {
+            return new TheoryData<EncryptingCredentialsTheoryData>
+            {
+                new EncryptingCredentialsTheoryData
+                {
+                    Key = null,
+                    Alg = null,
+                    Enc = null,
+                    ExpectedException = ExpectedException.ArgumentNullException("IDX10000:"),
+                    TestId = "null key"
+                },
+                new EncryptingCredentialsTheoryData
+                {
+                    Key = null,
+                    Alg = SecurityAlgorithms.RsaOaepKeyWrap,
+                    Enc = SecurityAlgorithms.Aes128CbcHmacSha256,
+                    ExpectedException = ExpectedException.ArgumentNullException("IDX10000:"),
```

Comment:
It is possible to check the exception contains "enc", this catches typo's in the error message.

Reply:
I'm going to use values in the following format as expected substrings:
*"IDX10000: The parameter enc"*